### ISSUE TITLE
Sending video attachments causes 500s 

### ISSUE THREAD

**Issue by jhankin** [Raheem-ai/seed_web#46]
When making a report from the chatbot, you can attach media and the chatbot will send it via `POST /attachments` to the backend.  Still images work correctly, but videos get dropped on the floor with 500 errors.  Example logs are below:

```
2017-03-26T21:18:24.487973+00:00 app[web.1]: I, [2017-03-26T21:18:24.487919 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] Started POST "/api/v1/attachments?incident_id=813&asset_url=https%3A%2F%2Fvideo.xx.fbcdn.net%2Fv%2Ft42.3356-2%2F17590350_10101361528133511_4008746101254389760_n.mp4%2Fvideo-1490563103.mp4%3Fvabr%3D492977%26oh%3Df379d4a6b8ba3dd0dc1b389dd048ee0c%26oe%3D58DA0590&write_key=[FILTERED]" for 54.159.46.36 at 2017-03-26 21:18:24 +0000
2017-03-26T21:18:24.489051+00:00 app[web.1]: I, [2017-03-26T21:18:24.488999 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] Processing by API::V1::AttachmentsController#create as JSON
2017-03-26T21:18:24.489053+00:00 app[web.1]: D, [2017-03-26T21:18:24.489005 #7] DEBUG -- : [5000f273-fc81-4f6b-8e4b-07cc7460dcda]   Role Load (1.8ms)  SELECT  "roles".* FROM "roles" WHERE "roles"."name" = $1 AND "roles"."resource_type" IS NULL AND "roles"."resource_id" IS NULL ORDER BY "roles"."id" ASC LIMIT $2  [["name", "normal"], ["LIMIT", 1]]
2017-03-26T21:18:24.489101+00:00 app[web.1]: I, [2017-03-26T21:18:24.489061 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a]   Parameters: {"incident_id"=>"813", "asset_url"=>"https://video.xx.fbcdn.net/v/t42.3356-2/17590350_10101361528133511_4008746101254389760_n.mp4/video-1490563103.mp4?vabr=492977&oh=f379d4a6b8ba3dd0dc1b389dd048ee0c&oe=58DA0590", "write_key"=>"[FILTERED]"}
2017-03-26T21:18:24.491689+00:00 app[web.1]: D, [2017-03-26T21:18:24.491639 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a]   APIKey Exists (1.5ms)  SELECT  1 AS one FROM "api_keys" WHERE "api_keys"."write_key" = $1 LIMIT $2  [["write_key", "1553dfb1fed95eecf4cfe2511bec1117"], ["LIMIT", 1]]
2017-03-26T21:18:24.492609+00:00 app[web.1]: D, [2017-03-26T21:18:24.492559 #7] DEBUG -- : [5000f273-fc81-4f6b-8e4b-07cc7460dcda]   Role Exists (1.8ms)  SELECT  1 AS one FROM "roles" INNER JOIN "users_roles" ON "roles"."id" = "users_roles"."role_id" WHERE "users_roles"."user_id" = $1 AND "roles"."id" = $2 LIMIT $3  [["user_id", 192], ["id", 1], ["LIMIT", 1]]
2017-03-26T21:18:24.504970+00:00 app[web.1]: D, [2017-03-26T21:18:24.504910 #7] DEBUG -- : [5000f273-fc81-4f6b-8e4b-07cc7460dcda]    (1.6ms)  COMMIT
2017-03-26T21:18:24.509305+00:00 app[web.1]: D, [2017-03-26T21:18:24.509249 #7] DEBUG -- : [5000f273-fc81-4f6b-8e4b-07cc7460dcda] [active_model_serializers]   Attachment Load (1.7ms)  SELECT "attachments".* FROM "attachments" WHERE "attachments"."incident_id" = $1  [["incident_id", 813]]
2017-03-26T21:18:24.514677+00:00 app[web.1]: D, [2017-03-26T21:18:24.514621 #7] DEBUG -- : [5000f273-fc81-4f6b-8e4b-07cc7460dcda] [active_model_serializers]   Officer Load (1.9ms)  SELECT "officers".* FROM "officers" WHERE "officers"."incident_id" = $1  [["incident_id", 813]]
2017-03-26T21:18:24.515345+00:00 app[web.1]: I, [2017-03-26T21:18:24.515295 #7]  INFO -- : [5000f273-fc81-4f6b-8e4b-07cc7460dcda] [active_model_serializers] Rendered IncidentSerializer with ActiveModelSerializers::Adapter::Json (9.75ms)
2017-03-26T21:18:24.515546+00:00 app[web.1]: I, [2017-03-26T21:18:24.515501 #7]  INFO -- : [5000f273-fc81-4f6b-8e4b-07cc7460dcda] Completed 200 OK in 50ms (Views: 6.8ms | ActiveRecord: 21.1ms)
2017-03-26T21:18:24.640108+00:00 app[web.1]: D, [2017-03-26T21:18:24.640032 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] DRAGONFLY: shell command: 'identify' '-ping' '-format' '%m %w %h' '/tmp/dragonfly20170326-11-nittyh.mp4'
2017-03-26T21:18:24.660122+00:00 app[web.1]: W, [2017-03-26T21:18:24.660048 #11]  WARN -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] DRAGONFLY: setting magic attribute for width to nil in <Dragonfly Attachment uid=nil, app=:default> because got error Command failed ('identify' '-ping' '-format' '%m %w %h' '/tmp/dragonfly20170326-11-nittyh.mp4') with exit status 1 and stderr identify.im6: delegate failed `"ffmpeg" -v -1 -i "%i" -vframes %S -vcodec pam -an -f rawvideo -y "%u.pam" 2> "%Z"' @ error/delegate.c/InvokeDelegate/1065.
2017-03-26T21:18:24.660124+00:00 app[web.1]: identify.im6: unable to open image `/tmp/magick-VGW80hgM.pam': No such file or directory @ error/blob.c/OpenBlob/2641.
2017-03-26T21:18:24.660125+00:00 app[web.1]: 
2017-03-26T21:18:24.660289+00:00 app[web.1]: D, [2017-03-26T21:18:24.660240 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] DRAGONFLY: shell command: 'identify' '-ping' '-format' '%m %w %h' '/tmp/dragonfly20170326-11-nittyh.mp4'
2017-03-26T21:18:24.535787+00:00 heroku[router]: at=info method=PATCH path="/api/v1/incidents/813?user_id=192&write_key=1553dfb1fed95eecf4cfe2511bec1117&incident_type_name=citation&start_time=1490400000&rating=1&description=I%20felt%20like%20a%20pool%20of%20cold%20water%20with%20one%20leaf%20floating%20in%20it&latitude=37.748345656973&longitude=-122.17693612467&metadata=%7B%22DATE_OF_INCIDENT%22%3A1490400000%2C%22TIME_OF_INCIDENT%22%3A%225pm%22%2C%22ENCOUNTER_LOCATION%22%3A%22%7B%5C%22lat%5C%22%3A37.748345656973%2C%5C%22long%5C%22%3A-122.17693612467%7D%22%2C%22CONTACT_OUTCOME%22%3A%22citation%22%2C%22CASE_NUMBER%22%3A%229462%22%2C%22INITIAL_CONTACT%22%3A%22got%20stopped%20on%20foot%22%2C%22POLICE_ACTION%22%3A%22other%22%2C%22POLICE_SEARCH%3F%22%3A%22yes%22%2C%22PERMISSION_TO_SEARCH%3F%22%3A%22no%22%2C%22NUMBER_OF_OFFICERS%22%3A%224%22%2C%22OFFICER_NAME%22%3A%22Hambone%20McGillicutty%22%2C%22OFFICER_BADGE_NUMBER%22%3A%226789%22%2C%22OFFICER_RACE%22%3A%22white%2Fcaucasian%22%2C%22OFFICER_GENDER%22%3A%22man%22%2C%22OFFICER_AGE%22%3A%2241-50%22%2C%22OFFICER_DETAILS_DESCRIPTION%22%3A%229%20feet%20tall%2C%20mouth%20full%20of%20corned%20beef%22%2C%22ENCOUNTER_SENTIMENT%22%3A%221%22%2C%22FURTHER_DESCRIPTION%22%3A%22I%20felt%20like%20a%20pool%20of%20cold%20water%20with%20one%20leaf%20floating%20in%20it%22%2C%22USER_REACTION%22%3A%22embarrassed%22%2C%22OFFICER_DISPOSITION%22%3A%22discriminatory%22%7D&reactions_list%5B0%5D=embarrassed&reactions_list%5B1%5D=discriminatory&tags_list%5B0%5D=citation" host=www.raheem.ai request_id=5000f273-fc81-4f6b-8e4b-07cc7460dcda fwd="54.159.46.36" dyno=web.1 connect=1ms service=56ms status=200 bytes=2271 protocol=https
2017-03-26T21:18:24.667315+00:00 app[web.1]: W, [2017-03-26T21:18:24.667260 #11]  WARN -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] DRAGONFLY: setting magic attribute for height to nil in <Dragonfly Attachment uid=nil, app=:default> because got error Command failed ('identify' '-ping' '-format' '%m %w %h' '/tmp/dragonfly20170326-11-nittyh.mp4') with exit status 1 and stderr identify.im6: delegate failed `"ffmpeg" -v -1 -i "%i" -vframes %S -vcodec pam -an -f rawvideo -y "%u.pam" 2> "%Z"' @ error/delegate.c/InvokeDelegate/1065.
2017-03-26T21:18:24.667325+00:00 app[web.1]: identify.im6: unable to open image `/tmp/magick-JtzxxIiQ.pam': No such file or directory @ error/blob.c/OpenBlob/2641.
2017-03-26T21:18:24.667325+00:00 app[web.1]: 
2017-03-26T21:18:24.669410+00:00 app[web.1]: D, [2017-03-26T21:18:24.669360 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a]    (1.5ms)  BEGIN
2017-03-26T21:18:24.672495+00:00 app[web.1]: D, [2017-03-26T21:18:24.672445 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a]   Incident Load (1.8ms)  SELECT  "incidents".* FROM "incidents" WHERE "incidents"."id" = $1 LIMIT $2  [["id", 813], ["LIMIT", 1]]
2017-03-26T21:18:24.740125+00:00 app[web.1]: D, [2017-03-26T21:18:24.740054 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a]    (1.4ms)  ROLLBACK
2017-03-26T21:18:24.740365+00:00 app[web.1]: I, [2017-03-26T21:18:24.740319 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] Completed 500 Internal Server Error in 251ms (ActiveRecord: 6.2ms)
2017-03-26T21:18:24.740459+00:00 app[web.1]: D, [2017-03-26T21:18:24.740418 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] [Rollbar] Reporting exception: Invalid image file
2017-03-26T21:18:24.768842+00:00 app[web.1]: I, [2017-03-26T21:18:24.768780 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] [Rollbar] Scheduling item
2017-03-26T21:18:24.768874+00:00 app[web.1]: I, [2017-03-26T21:18:24.768838 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] [Rollbar] Sending item
2017-03-26T21:18:25.148719+00:00 heroku[router]: at=info method=POST path="/api/v1/attachments?incident_id=813&asset_url=https%3A%2F%2Fvideo.xx.fbcdn.net%2Fv%2Ft42.3356-2%2F17590350_10101361528133511_4008746101254389760_n.mp4%2Fvideo-1490563103.mp4%3Fvabr%3D492977%26oh%3Df379d4a6b8ba3dd0dc1b389dd048ee0c%26oe%3D58DA0590&write_key=1553dfb1fed95eecf4cfe2511bec1117" host=www.raheem.ai request_id=31e84a1e-67ac-4928-94e1-b8ac71f5fa0a fwd="54.159.46.36" dyno=web.1 connect=1ms service=662ms status=500 bytes=243 protocol=https
2017-03-26T21:18:25.144861+00:00 app[web.1]: W, [2017-03-26T21:18:25.144765 #11]  WARN -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] [Rollbar] Got unexpected status code from Rollbar api: 429
2017-03-26T21:18:25.144913+00:00 app[web.1]: I, [2017-03-26T21:18:25.144851 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] [Rollbar] Response: {
2017-03-26T21:18:25.144914+00:00 app[web.1]:   "err": 1,
2017-03-26T21:18:25.144915+00:00 app[web.1]:   "message": "Over plan monthly limit. Project id 89859 access token f9cb01305bcf49e4bf4fc42d0ca39338."
2017-03-26T21:18:25.144915+00:00 app[web.1]: }
2017-03-26T21:18:25.144968+00:00 app[web.1]: I, [2017-03-26T21:18:25.144927 #11]  INFO -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] [Rollbar] Details: https://rollbar.com/instance/uuid?uuid=6c76efac-9f30-4921-8c13-ece1b32289eb (only available if report was successful)
2017-03-26T21:18:25.145023+00:00 app[web.1]: D, [2017-03-26T21:18:25.144983 #11] DEBUG -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] [Rollbar] Exception uuid saved in env: 6c76efac-9f30-4921-8c13-ece1b32289eb
2017-03-26T21:18:25.145800+00:00 app[web.1]: F, [2017-03-26T21:18:25.145753 #11] FATAL -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a]   
2017-03-26T21:18:25.145840+00:00 app[web.1]: F, [2017-03-26T21:18:25.145802 #11] FATAL -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] CloudinaryException (Invalid image file):
2017-03-26T21:18:25.145874+00:00 app[web.1]: F, [2017-03-26T21:18:25.145839 #11] FATAL -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a]   
2017-03-26T21:18:25.145938+00:00 app[web.1]: F, [2017-03-26T21:18:25.145878 #11] FATAL -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] app/models/cloudinary_store.rb:5:in `write'
2017-03-26T21:18:25.145972+00:00 app[web.1]: F, [2017-03-26T21:18:25.145936 #11] FATAL -- : [31e84a1e-67ac-4928-94e1-b8ac71f5fa0a] app/controllers/api/v1/attachments_controller.rb:5:in `create'
```

**Comment by kidbombay** [comment]
Let's add a static video url test to the api spec to make sure this works there first.
We haven't discussed what video formats we will support either.

**Comment by kidbombay** [comment]
This bug was occurring because server side default for cloudinary is to set all uploads resource type to image. Dragonfly was then trying to process the file as image instead of video.

The fix in #72 sets resource type to auto without any other changes. All tests are passing.
